plugins/by-name/neorg/default.nix

```diff
@@ -1,173 +1,249 @@
 {
   lib,
-  helpers,
   config,
   pkgs,
   ...
 }:
 let
-  cfg = config.plugins.neorg;
+  inherit (lib.nixvim)
+    defaultNullOpts
+    mkNullOrStr
+    mkNullOrOption
+    mkNullOrOption'
+    ;
+  inherit (lib) types;
 in
-with lib;
-{
-  options.plugins.neorg = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "neorg";
-
-    package = lib.mkPackageOption pkgs "neorg" {
-      default = [
-        "vimPlugins"
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "neorg";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  # TODO introduced 2024-12-12: remove after 25.05
```

Comment:
One thing we talked about recently was creating a separate `deprecations.nix` that we import next to each file in our `by-name` structure so that it doesn't pollute the main file. 

`plugins/by-name/neorg/deprecations.nix`

Reply:
Good point. I didn't move the `optionsRenamedToSettings` though as it is less straightforward.